R/registry.R

```diff
@@ -513,3 +513,69 @@ transition_model_version_stage <- function(name, version, stage, archive_existin
 
   return(response$model_version)
 }
+
+#' Gather metadata for a model version
+#'
+#' Model versions are tied to experiments, runs, and registered model names.
+#' This helper returns metadata on the experiment and run that are associated with the registered model name and version provided.
+#'
+#' @param registered_model_name The name of the registered model
+#' @param version The version of the model for which to gather metadata
+#' @param client An MLFlow client. Autogenerated if not provided
+#'
+#' @return A list containing the `registered_model_name`, the `experiment_id`, the `experiment_name`, the `run_id`, and the `model_version`
+#' @export
+gather_model_metadata <- function(registered_model_name, version, client = mlflow_client()) {
+  model_version_meta <- get_model_version(
+    name = registered_model_name,
+    version = version,
+    client = client
+  )
+
+  experiment_id <- get_experiment_from_run(run_id = model_version_meta$run_id)
+  experiment_name <- get_experiment(experiment_id = experiment_id, client = client)$name
+  model_version <- model_version_meta$version
+  run_id <- model_version_meta$run_id
+
+  list(
+    registered_model_name = registered_model_name,
+    experiment_id = experiment_id,
+    experiment_name = experiment_name,
+    run_id = run_id,
+    model_version = model_version
+  )
```

Comment:
is there a reason to not return a one-row df? i guess there's really no difference, and why use a df if there aren't multiple rows

Reply:
Yeah I feel like a list is safer, because then we can map over multiple things and return a list of lists instead of a list of dfs